 Network rinha-nginx-2024q1  Creating
 Network rinha-nginx-2024q1  Created
 Container devkemc-node-db-1  Creating
 Container devkemc-node-db-1  Created
 Container devkemc-node-api02-1  Creating
 Container devkemc-node-api01-1  Creating
 Container devkemc-node-api02-1  Created
 Container devkemc-node-api01-1  Created
 Container devkemc-node-nginx-1  Creating
 Container devkemc-node-nginx-1  Created
Attaching to api01-1, api02-1, db-1, nginx-1
db-1     | The files belonging to this database system will be owned by user "postgres".
db-1     | This user must also own the server process.
db-1     | 
db-1     | The database cluster will be initialized with locale "en_US.utf8".
db-1     | The default database encoding has accordingly been set to "UTF8".
db-1     | The default text search configuration will be set to "english".
db-1     | 
db-1     | Data page checksums are disabled.
db-1     | 
db-1     | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db-1     | creating subdirectories ... ok
db-1     | selecting dynamic shared memory implementation ... posix
db-1     | selecting default max_connections ... 100
nginx-1  | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
nginx-1  | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
nginx-1  | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
nginx-1  | 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
nginx-1  | 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
nginx-1  | /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
nginx-1  | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
nginx-1  | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
nginx-1  | /docker-entrypoint.sh: Configuration complete; ready for start up
db-1     | selecting default shared_buffers ... 128MB
api02-1  | {"level":30,"time":1707854076830,"pid":1,"hostname":"app2","msg":"server.js:1:Listening on 8080"}
api01-1  | {"level":30,"time":1707854076830,"pid":1,"hostname":"api01","msg":"server.js:1:Listening on 8080"}
db-1     | selecting default time zone ... Etc/UTC
db-1     | creating configuration files ... ok
db-1     | running bootstrap script ... ok
api01-1  | {"level":50,"time":1707854079143,"pid":1,"hostname":"api01","err":{"type":"Error","message":"connect ECONNREFUSED 172.20.0.2:5432","stack":"Error: connect ECONNREFUSED 172.20.0.2:5432\n    at /usr/src/app/node_modules/pg-pool/index.js:45:11\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findById (file:///usr/src/app/database.js:79:20)\n    at async getExtrato (file:///usr/src/app/server.js:70:21)","errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"172.20.0.2","port":5432},"msg":"connect ECONNREFUSED 172.20.0.2:5432"}
api01-1  | {"level":50,"time":1707854081174,"pid":1,"hostname":"api01","err":{"type":"Error","message":"connect ECONNREFUSED 172.20.0.2:5432","stack":"Error: connect ECONNREFUSED 172.20.0.2:5432\n    at /usr/src/app/node_modules/pg-pool/index.js:45:11\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findById (file:///usr/src/app/database.js:79:20)\n    at async getExtrato (file:///usr/src/app/server.js:70:21)","errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"172.20.0.2","port":5432},"msg":"connect ECONNREFUSED 172.20.0.2:5432"}
api01-1  | {"level":50,"time":1707854083185,"pid":1,"hostname":"api01","err":{"type":"Error","message":"connect ECONNREFUSED 172.20.0.2:5432","stack":"Error: connect ECONNREFUSED 172.20.0.2:5432\n    at /usr/src/app/node_modules/pg-pool/index.js:45:11\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findById (file:///usr/src/app/database.js:79:20)\n    at async getExtrato (file:///usr/src/app/server.js:70:21)","errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"172.20.0.2","port":5432},"msg":"connect ECONNREFUSED 172.20.0.2:5432"}
db-1     | performing post-bootstrap initialization ... ok
db-1     | syncing data to disk ... ok
db-1     | 
db-1     | 
db-1     | Success. You can now start the database server using:
db-1     | 
db-1     |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db-1     | 
db-1     | initdb: warning: enabling "trust" authentication for local connections
db-1     | initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
db-1     | waiting for server to start....2024-02-13 19:54:44.498 UTC [48] LOG:  starting PostgreSQL 16.2 (Debian 16.2-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
db-1     | 2024-02-13 19:54:44.523 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db-1     | 2024-02-13 19:54:44.593 UTC [51] LOG:  database system was shut down at 2024-02-13 19:54:43 UTC
db-1     | 2024-02-13 19:54:44.611 UTC [48] LOG:  database system is ready to accept connections
db-1     |  done
db-1     | server started
api01-1  | {"level":50,"time":1707854085197,"pid":1,"hostname":"api01","err":{"type":"Error","message":"connect ECONNREFUSED 172.20.0.2:5432","stack":"Error: connect ECONNREFUSED 172.20.0.2:5432\n    at /usr/src/app/node_modules/pg-pool/index.js:45:11\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findById (file:///usr/src/app/database.js:79:20)\n    at async getExtrato (file:///usr/src/app/server.js:70:21)","errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"172.20.0.2","port":5432},"msg":"connect ECONNREFUSED 172.20.0.2:5432"}
db-1     | CREATE DATABASE
db-1     | 
db-1     | 
db-1     | /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/script.sql
api02-1  | {"level":50,"time":1707854087223,"pid":1,"hostname":"app2","err":{"type":"Error","message":"connect ECONNREFUSED 172.20.0.2:5432","stack":"Error: connect ECONNREFUSED 172.20.0.2:5432\n    at /usr/src/app/node_modules/pg-pool/index.js:45:11\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async findById (file:///usr/src/app/database.js:79:20)\n    at async getExtrato (file:///usr/src/app/server.js:70:21)","errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"172.20.0.2","port":5432},"msg":"connect ECONNREFUSED 172.20.0.2:5432"}
db-1     | CREATE TABLE
db-1     | CREATE TABLE
db-1     | CREATE INDEX
db-1     | INSERT 0 5
db-1     | 
db-1     | 
db-1     | 2024-02-13 19:54:47.352 UTC [48] LOG:  received fast shutdown request
db-1     | waiting for server to shut down....2024-02-13 19:54:47.357 UTC [48] LOG:  aborting any active transactions
db-1     | 2024-02-13 19:54:47.393 UTC [48] LOG:  background worker "logical replication launcher" (PID 54) exited with exit code 1
db-1     | 2024-02-13 19:54:47.494 UTC [49] LOG:  shutting down
db-1     | 2024-02-13 19:54:47.503 UTC [49] LOG:  checkpoint starting: shutdown immediate
db-1     | 2024-02-13 19:54:47.974 UTC [49] LOG:  checkpoint complete: wrote 939 buffers (5.7%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.425 s, sync=0.035 s, total=0.481 s; sync files=308, longest=0.011 s, average=0.001 s; distance=4297 kB, estimate=4297 kB; lsn=0/191D1D8, redo lsn=0/191D1D8
db-1     | 2024-02-13 19:54:47.995 UTC [48] LOG:  database system is shut down
db-1     |  done
db-1     | server stopped
db-1     | 
db-1     | PostgreSQL init process complete; ready for start up.
db-1     | 
db-1     | 2024-02-13 19:54:48.296 UTC [1] LOG:  starting PostgreSQL 16.2 (Debian 16.2-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
db-1     | 2024-02-13 19:54:48.296 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db-1     | 2024-02-13 19:54:48.296 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db-1     | 2024-02-13 19:54:48.340 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db-1     | 2024-02-13 19:54:48.394 UTC [66] LOG:  database system was shut down at 2024-02-13 19:54:47 UTC
db-1     | 2024-02-13 19:54:48.434 UTC [1] LOG:  database system is ready to accept connections
api01-1  | {"level":30,"time":1707854089288,"pid":1,"hostname":"api01","msg":"database.js: Connected  to db postgres://admin:123@db:5432/rinha"}
api02-1  | {"level":30,"time":1707854089321,"pid":1,"hostname":"app2","msg":"database.js: Connected  to db postgres://admin:123@db:5432/rinha"}
nginx-1  | 2024/02/13 19:56:57 [alert] 1#1: worker process 29 exited on signal 9
nginx-1  | 2024/02/13 19:57:04 [alert] 1#1: worker process 30 exited on signal 9
nginx-1  | 2024/02/13 19:57:11 [alert] 1#1: worker process 31 exited on signal 9
nginx-1  | 2024/02/13 19:57:19 [alert] 1#1: worker process 32 exited on signal 9
nginx-1  | 2024/02/13 19:57:27 [alert] 1#1: worker process 33 exited on signal 9
nginx-1  | 2024/02/13 19:57:35 [alert] 1#1: worker process 34 exited on signal 9
nginx-1  | 2024/02/13 19:57:42 [alert] 1#1: worker process 35 exited on signal 9
nginx-1  | 2024/02/13 19:57:50 [alert] 1#1: worker process 36 exited on signal 9
nginx-1  | 2024/02/13 19:57:58 [alert] 1#1: worker process 37 exited on signal 9
nginx-1  | 2024/02/13 19:58:06 [alert] 1#1: worker process 38 exited on signal 9
nginx-1  | 2024/02/13 19:58:13 [alert] 1#1: worker process 39 exited on signal 9
nginx-1  | 2024/02/13 19:58:21 [alert] 1#1: worker process 40 exited on signal 9
nginx-1  | 2024/02/13 19:58:29 [alert] 1#1: worker process 41 exited on signal 9
nginx-1  | 2024/02/13 19:58:36 [alert] 1#1: worker process 42 exited on signal 9
nginx-1  | 2024/02/13 19:58:44 [alert] 1#1: worker process 43 exited on signal 9
nginx-1  | 2024/02/13 19:58:52 [alert] 1#1: worker process 44 exited on signal 9
nginx-1  | 2024/02/13 19:58:59 [alert] 1#1: worker process 45 exited on signal 9
db-1     | 2024-02-13 19:59:48.494 UTC [64] LOG:  checkpoint starting: time
nginx-1 exited with code 0
nginx-1 exited with code 0
api02-1  | /usr/src/app/node_modules/pg-pool/index.js:45
api02-1  |     Error.captureStackTrace(err);
api02-1  |           ^
api02-1  | 
api02-1  | Error: timeout exceeded when trying to connect
api02-1  |     at /usr/src/app/node_modules/pg-pool/index.js:45:11
api02-1  |     at async updateClient (file:///usr/src/app/database.js:49:18)
api02-1  |     at async IncomingMessage.<anonymous> (file:///usr/src/app/server.js:56:23)
api02-1  | 
api02-1  | Node.js v21.6.1
db-1     | 2024-02-13 20:00:14.021 UTC [71] LOG:  could not receive data from client: Connection reset by peer
api02-1 exited with code 0
api02-1 exited with code 1
api01-1 exited with code 0
api01-1 exited with code 137
db-1     | 2024-02-13 20:00:16.965 UTC [1] LOG:  received fast shutdown request
db-1     | 2024-02-13 20:00:16.967 UTC [1] LOG:  aborting any active transactions
db-1     | 2024-02-13 20:00:16.995 UTC [1] LOG:  background worker "logical replication launcher" (PID 69) exited with exit code 1
db-1     | 2024-02-13 20:00:17.032 UTC [64] LOG:  checkpoint complete: wrote 548 buffers (3.3%); 0 WAL file(s) added, 0 removed, 1 recycled; write=28.500 s, sync=0.013 s, total=28.539 s; sync files=29, longest=0.009 s, average=0.001 s; distance=15467 kB, estimate=15467 kB; lsn=0/29CB858, redo lsn=0/2837F88
db-1     | 2024-02-13 20:00:17.032 UTC [64] LOG:  shutting down
db-1     | 2024-02-13 20:00:17.034 UTC [64] LOG:  checkpoint starting: shutdown immediate
db-1     | 2024-02-13 20:00:17.059 UTC [64] LOG:  checkpoint complete: wrote 55 buffers (0.3%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.008 s, sync=0.010 s, total=0.028 s; sync files=8, longest=0.004 s, average=0.002 s; distance=1614 kB, estimate=14082 kB; lsn=0/29CB8D0, redo lsn=0/29CB8D0
db-1     | 2024-02-13 20:00:17.098 UTC [1] LOG:  database system is shut down
db-1 exited with code 0
